main.c

```diff
@@ -783,6 +789,9 @@ static int parse_args(int argc, char *argv[], char **config_path) {
 	while ((c = getopt(argc, argv, "C:hdwS:")) != -1) {
 		switch (c) {
 		case 'C':
+			if (*config_path != NULL) {
+				free(*config_path);
```

Comment:
FYI, you don't need the null check here around `free`: ["If ptr is a null pointer, no action shall  occur."](https://man7.org/linux/man-pages/man3/free.3p.html).

Reply:
Very true. I don't know what I was thinking when I added that check :)